src/components/render-settings-panel.js

```diff
@@ -486,6 +365,71 @@ class RenderSettingsPanel extends Component {
+      adapter.scene.animationLoop.timeline.detachAnimation(this.state.cameraHandle)
+      adapter.scene.keyframes = this.resetKeyframes() // Resets keyframes so that they don't inherit other options
+      console.log("DETACHED")
+    }
+
+    this.parseSetCameraType(strCameraType, adapter.scene.keyframes.camera)
+
+    const newCameraHandle = adapter.scene.animationLoop.timeline.attachAnimation(adapter.scene.keyframes.camera);
+    this.setState({cameraHandle: newCameraHandle})
+  }
+
+  parseSetCameraType(strCameraType, camera) {
+    const match = strCameraType.match(/\b(?!to)\b\S+\w/g) // returns arr of important keywords. Should work for 2+ words in future ex: ["Orbit", "90"] | ["North", "South"] | ["Zoom", "In"]
+
+    // Named this way for possibility of >2 keyframes in future
+    const firstKeyframe = camera.values[0]
+    const secondKeyframe = camera.values[1]
+
+    // Converts mapState object to turf friendly Point obj (GEOJSON)
+    const point = turf.point([camera.values[1].longitude, camera.values[1].latitude])
+    
+    if (match[0] == "Orbit") {
+      secondKeyframe.bearing = parseInt(match[1])
+    }
+
+    // TODO future option that'll allow user to set X distance (km OR miles) directionally. Options inside turf
+    // https://turfjs.org/docs/#transformTranslate
+    const set_checker = new Set(["East", "South", "West", "North"])
+    if (set_checker.has(match[0])) {
+      if (match[0] == "East") { // TODO Temporary solution to catch this branch to master. Doesn't work for "East to North" for example if option allows in future
+        const translatedPoly = turf.transformTranslate(point, 10000, 270);
+        secondKeyframe.longitude = translatedPoly.geometry.coordinates[0]
+      } else 
+      if (match[0] == "South") {
+        const translatedPoly = turf.transformTranslate(point, 10000, 0);
+        secondKeyframe.latitude = translatedPoly.geometry.coordinates[1]
+      } else 
+      if (match[0] == "West") {
+        const translatedPoly = turf.transformTranslate(point, 10000, 90);
+        secondKeyframe.longitude = translatedPoly.geometry.coordinates[0]
+      } else 
+      if (match[0] == "North") {
+        const translatedPoly = turf.transformTranslate(point, 10000, 180);
+        secondKeyframe.latitude = translatedPoly.geometry.coordinates[1]
+      }
+    }
+
+    if (match[0] == "Zoom") {
+      if (match[1] == "In") {
+        secondKeyframe.zoom = 15
+      } else 
+      if (match[1] == "Out") {
+        firstKeyframe.zoom = 15
+      } 
+    }
+  }
+
+  resetKeyframes() { // minified default keyframes from scenebuilder fn
+    return {camera:new CameraKeyframes({timings:[0,1000],keyframes:[{longitude:mapdataGlobal.mapState.longitude,latitude:mapdataGlobal.mapState.latitude,zoom:mapdataGlobal.mapState.zoom,pitch:0,bearing:0},{longitude:mapdataGlobal.mapState.longitude,latitude:mapdataGlobal.mapState.latitude,zoom:mapdataGlobal.mapState.zoom,bearing:0,pitch:0}],easings:[easing.easeInOut]})}
```

Comment:
Let’s make this more legible on multiple lines (legibility over short files always since minifcation will take care of any concerns one has)

Reply:
This is temporary and will be removed once we implement an initialViewstate in Hubble. Looks like Elisa is handling that this week so it'll be removed afterwards